packages/flutter_hooks/lib/src/change_notifier.dart

```diff
@@ -0,0 +1,30 @@
+part of 'hooks.dart';
+
+/// Creates a general purpose hook on a [ChangeNotifier] subclass that will be
+/// disposed automatically.
+///
+/// For example, these are equivalent:
+/// ```dart
+/// useTextEditingController(text: 'Hello');
+/// ```
+/// And:
+/// ```dart
+/// useDisposable(() => TextEditingController(text: 'Hello'));
+/// ```
+/// This is very useful if you can't find a "common controller" class in Flutter
+/// which has not a hook implementation, for example `CupertinoTabController`:
+/// ```dart
+/// useDisposable(() => CupertinoTabController());
+/// ```
+T useChangeNotifier<T extends ChangeNotifier>(
+  T Function() factory, [
+  String? label,
```

Comment:
I don't like this being part of the public API

Reply:
as I commented, it can be used to hook a common ChangeNotifier that has not hook yet, for example `CupertinoTabController` which was expected by some people: `[CupertinoTabController](https://api.flutter.dev/flutter/cupertino/CupertinoTabController-class.html)`